src/nfts/index.ts

```diff
@@ -23,20 +24,28 @@ export enum NftVendor {
 
 export const NftVendorKeys = Object.values(NftVendor);
 
-export type NftDetails = AssetDetail & { assetId: string };
-
 export interface BaseInfo {
   id: string;
   vendor: NftVendor;
 }
 
 export class BaseNft<TypedInfo extends BaseInfo> {
   protected asset: AssetDetail;
-  protected info: TypedInfo | null;
+  protected info: TypedInfo | null | undefined;
+  protected config: NftConfig | null | undefined;
```

Comment:
If prop type matches what it's assigned inside constructor, you can omit types here.